src/lib.rs

```diff
@@ -97,3 +104,67 @@ fn init_logging() {
         }
     }
 }
+
+///Returns true iff a valid port configuration for the cmdline command exists optionally printing error messages for the user
+fn check_valid_port_available(print_error_message: bool) -> bool {
+    let host = "127.0.0.1";
+
+    // Check default port
+    let default_port_valid = TcpPort::is_port_available(host, DEFAULT_PORT);
+
+    // Check port set via env var SCCACHE_SERVER_PORT
+    let mut env_port_valid = false;
+    let mut env_port: u16 = 0;
+
+    let env_var_raw = env::var("SCCACHE_SERVER_PORT")
+        .ok()
+        .and_then(|s| s.parse().ok());
+
+    if env_var_raw.is_some() {
+        env_port = env_var_raw.unwrap();
+        env_port_valid = TcpPort::is_port_available(host, env_port);
+    }
+
+    // If neither are valid for binding a new server
+    if !env_port_valid && !default_port_valid {
+        let cmdline_command = cmdline::parse();
+
+        if cmdline_command.is_ok() {
+            let cmd = cmdline_command.unwrap();
+            
+            // and we are running a --start-server 
+            if cmd == cmdline::Command::StartServer {
```

Comment:
FWIW, it's possible to do this without `PartialEq` and also without a match by doing:
```suggestion
            if let cmdline::Command::StartServer = cmd { ... }
```
I believe :)